source/common/quic/envoy_quic_server_preferred_address_config_factory.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "envoy/config/typed_config.h"
+#include "envoy/network/address.h"
+#include "envoy/protobuf/message_validator.h"
+
+#include "quiche/quic/platform/api/quic_socket_address.h"
+
+namespace Envoy {
+namespace Quic {
+
+class EnvoyQuicServerPreferredAddressConfig {
+public:
+  virtual ~EnvoyQuicServerPreferredAddressConfig() = default;
+
+  /**
+   * @param local_address the configured default listening address.
+   * Returns a pair of the server preferred addresses in form of {IPv4, IPv6}.
```

Comment:
What happens if the server only supports v4 or only v6? Is one of the addresses empty?